database/migrations/2019_03_08_204743_adds_callpower_campaign_id_unique_index_in_actions_table.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddsCallpowerCampaignIdUniqueIndexInActionsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('actions', function (Blueprint $table) {
+            $table->index('callpower_campaign_id');
```

Comment:
Since we ensure that this field is unique in our validation rules, we can use a [unique index](https://laravel.com/docs/5.8/migrations#indexes) here.

Reply:
ah yes - updated!